### PR TITLE
Add Reclaim.ai extension

### PR DESCRIPTION
**Pull request by webergnr**
## Description

The reclaim extension helps you to use some of the most used reclaim functionalities without the need to go to a web browser. This initial version adds the following commands:

- Quick Task (simply create a task)
- Create Task (add a new task with more details)
- Scheduling links (List your current scheduling links)
- My Calendar (show your events)

This version also adds a menubar that allows the user to see at a glance what is happening now and what are the upcoming events on the list.

#### Onboard process

https://github.com/raycast/extensions/assets/5560204/3e789457-9281-4996-a0d3-39868c23025f

#### Using the extension

https://github.com/raycast/extensions/assets/5560204/2481d32d-3ca4-43d9-bfa4-74a1a285d2b3

https://github.com/raycast/extensions/assets/5560204/f25fa4cc-9177-4f34-86f5-50b5f4c79825

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/reclaim/src/hooks/useApi.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import { fetcher } from "../utils/axiosPromise";
+
+const useApi = () => {
+  const { apiToken } = getPreferenceValues<NativePreferences>();
+
+  const { push } = useNavigation();
+
+  if (!apiToken) {
+    push(
+      <Detail markdown={"Something wrong with your API Token key. Check your raycast config and set up a new token."} />
```
Comment (peduarte) [inline extensions/reclaim/src/hooks/useApi.tsx:12]: Capitalised Raycast
```suggestion
      <Detail markdown={"Something wrong with your API Token key. Check your Raycast config and set up a new token."} />
```

Comment (peduarte) [reply extensions/reclaim/src/hooks/useApi.tsx:12]: You can also add some `actions` to this `Detail` that [opens the Preferences](https://developers.raycast.com/api-reference/preferences#opencommandpreferences)

```
<Action
      title="Open Preferences"
      onAction={openCommandPreferences}
/>
```

File: extensions/reclaim/src/hooks/useInterpreter.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  // const confirmInterpreterMessage = async (planUuid: string) => {
+  //   try {
+  //     const rawRequest = await fetcher(`/interpreter/plans/applied/${planUuid}`, {
+  //       method: "POST",
+  //     });
+  //     return { statusCode: rawRequest.status };
+  //   } catch (error) {
+  //     console.error("Error while confirming interpreter message", error);
+  //   }
+  // };
```
Comment (peduarte) [inline extensions/reclaim/src/hooks/useInterpreter.tsx:42]: want to keep these?

File: extensions/reclaim/src/types/user.ts [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+  //   habits: boolean;
+  //   tasks: boolean;
+  //   googleTasks: boolean;
+  //   planItemPrioritized: boolean;
+  //   smartOneOnOnes: boolean;
+  //   bufferTime: boolean;
+  //   tasksReindex: boolean;
+  //   googleAddOn: boolean;
+  //   schedulingLinks: boolean;
+  // };
```
Comment (peduarte) [inline extensions/reclaim/src/types/user.ts:163]: want to keep these?

File: extensions/reclaim/src/utils/emoji-helper.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export const emojiFindingRegex = () =>
```
Comment (peduarte) [inline extensions/reclaim/src/utils/emoji-helper.ts:1]: wow! 😄

File: extensions/reclaim/src/hooks/useApi.tsx (hunk 1 above)
Comment (webergnr) [reply extensions/reclaim/src/hooks/useApi.tsx]: Nice!

File: extensions/reclaim/src/types/user.ts (hunk 3 above)
Comment (webergnr) [reply extensions/reclaim/src/types/user.ts]: I'm also fixing this :)
(and some other commented types)

**Review by peduarte — APPROVED** [review]

**Pull request merged**